iidm/iidm-api/src/main/java/com/powsybl/iidm/network/NetworkListener.java

```diff
@@ -6,33 +6,40 @@
  */
 package com.powsybl.iidm.network;
 
+import com.powsybl.commons.extensions.Extension;
+
 /**
  * @author Geoffroy Jamgotchian {@literal <geoffroy.jamgotchian at rte-france.com>}
  */
 public interface NetworkListener {
 
-    void onCreation(Identifiable identifiable);
+    void onCreation(Identifiable<?> identifiable);
 
-    void beforeRemoval(Identifiable identifiable);
+    void beforeRemoval(Identifiable<?> identifiable);
 
     void afterRemoval(String id);
 
-    void onUpdate(Identifiable identifiable, String attribute, Object oldValue, Object newValue);
+    void onUpdate(Identifiable<?> identifiable, String attribute, Object oldValue, Object newValue);
 
-    default void onUpdate(Identifiable identifiable, String attribute, String variantId, Object oldValue,
-            Object newValue) {
-        // empty default implementation
-    }
```

Comment:
Why did you remove this default implementation but keep the ones for `onElementAdded`, `onElementReplaced` and `onElementRemoved`?

Reply:
We could also. I will open a PR to also rename them as element is not a right term (this is in really only properties add or remove)